crates/re_viewer/src/app.rs

```diff
@@ -572,12 +606,19 @@ impl eframe::App for App {
             self.pending_commands.push(cmd);
         }
 
-        self.run_pending_commands(egui_ctx, frame);
+        self.run_pending_commands(&mut blueprint, egui_ctx, frame);
 
         self.frame_time_history.add(
             egui_ctx.input(|i| i.time),
             frame_start.elapsed().as_secs_f32(),
         );
+
+        // If this wasn't the default blueprint, save it back
+        if this_frame_blueprint_id != Default::default() {
+            let blueprint_db = self.log_dbs.get_mut(&this_frame_blueprint_id).unwrap();
+
+            blueprint.process_updates(&blueprint_snapshot, blueprint_db);
```

Comment:
```suggestion
            blueprint.store_changes(&blueprint_snapshot, blueprint_db);
```

or something would be a more descriptive name imho